examples/units_and_coordinates/radec_to_hpc_map.py

```diff
@@ -0,0 +1,148 @@
+# type projection, which here is RA-DEC.
+print(header['ctype1'], header['ctype2'])
+
+###############################################################################
+# Lets pull out the observation time and wavelength from the header, we will use
+# these to create our new header.
+obstime = Time(header['date-obs'])
+frequency = header['crval3']*u.Hz
+
+
+###############################################################################
+# To create a new `~sunpy.map.Map` header we need convert the reference coordinate
+# in RA-DEC (that is in the header) to Helioprojective. To do this we will first create
+# an `astropy.coordinates.SkyCoord` of the reference coordinate from the header information.
+# We will need the location of the observer (i.e. where the observation was taken).
+# We first establish the location on Earth from which the observation takes place, in this
+# case LOFAR observations are taken from Exloo in the Netherlands, which we define in lat and lon.
+# We can convert this to a SkyCoord in GCRSat the observation time.
+
+lofar_loc = EarthLocation(lat=52.905329712*u.deg, lon=6.867996528*u.deg)
+lofar_gcrs = SkyCoord(lofar_loc.get_gcrs(obstime))
+
+##########################################################################
+# We can then define the reference coordinate in terms of RA-DEC from the header information.
+# Here we are using the `obsgeoloc` keyword argument to take into account that the observer is not
+# at the center of the Earth (i.e. the GCRS origin). The distance here is the Sun-observer distance.
+reference_coord = SkyCoord(header['crval1']*u.Unit(header['cunit1']),
+                           header['crval2']*u.Unit(header['cunit2']),
+                           frame='gcrs',
+                           obstime=obstime,
+                           obsgeoloc=lofar_gcrs.cartesian,
+                           obsgeovel=lofar_gcrs.velocity.to_cartesian(),
+                           distance=lofar_gcrs.hcrs.distance)
+
+##########################################################################
+# Now we can convert the `reference_coord` to the HPC coordinate frame
+reference_coord_arcsec = reference_coord.transform_to(frames.Helioprojective(observer=lofar_gcrs))
+
+##########################################################################
+# Now we need to get the other parameters from the header that will be used
+# to create the new header - here we can get the cdelt1 and cdelt2 which are
+# the spatial scales of the data axes.
+cdelt1 = (np.abs(header['cdelt1'])*u.deg).to(u.arcsec)
+cdelt2 = (np.abs(header['cdelt2'])*u.deg).to(u.arcsec)
+
+##################################################################################
+# We will also need knowledge of the P-angle at the observation time, which is the
+# position angle between geocentric north and solar north as observed from Earth.
+# The image will need to be rotated by this angle so that solar north is pointed up.
```

Comment:
```suggestion
##################################################################################
# Finally, we need to specify the orientation of the HPC coordinate grid because
# GCRS north is not in the same direction as HPC north.  For convenience, we use
# :func:`~sunpy.coordinates.sun.P` to calculate this relative rotation angle,
# although due to subtleties in definitions, the returned value is inaccurate by
# 2 arcmin, equivalent to a worst-case shift of 0.6 arcsec for HPC coordinates
# on the disk.
```